### PR TITLE
feat(api): Add max page size

### LINKED ISSUE #160: Add maximum page size in fetch endpoints
## Description
Currently, the endpoints that fetch paginated data does not provide any upper limit to the page size. We need to put a limit to that to make sure we don't burn our API.

## Solution
We would like to add an upper limit to these fetch endpoints, something like this:
```
-take: limit,
+take: Math.min(limit, 30)
```
The maximum items per page should be 30. 

All the controllers in all the modules that have a paginated response will be affected.

### PR DESCRIPTION
**Pull request by DhairyaMajmudar**
## Description
Adding max page size to 30

Fixes #160

## Dependencies
N/A

## Future Improvements
N/A

## Mentions
@rajdip-b 

## Screenshots of relevant screens
N/A

## Developer's checklist

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-check on my work

**If changes are made in the code:**

- [x] I have followed the [coding guidelines](https://google.github.io/styleguide/jsguide.html)
- [x] My changes in code generate no new warnings
- [ ] My changes are breaking another fix/feature of the project
- [ ] I have added test cases to show that my feature works
- [ ] I have added relevant screenshots in my PR
- [ ] There are no UI/UX issues

### Documentation Update

- [ ] This PR requires an update to the documentation at [docs.keyshade.xyz](https://docs.keyshade.xyz)
- [ ] I have made the necessary updates to the documentation, or no documentation changes are required.

### CODE REVIEW

File: apps/api/src/project/service/project.service.ts
Comment (rajdip-b) [inline apps/api/src/project/service/project.service.ts:1]: Fetching forks missing

**Review by rajdip-b — APPROVED** [review]

File: apps/api/src/project/service/project.service.ts
Comment (DhairyaMajmudar) [reply apps/api/src/project/service/project.service.ts:1]: Didn't get you pt. here

Comment (rajdip-b) [reply apps/api/src/project/service/project.service.ts:1]: I meant, there's one more endpoint which is used to fetch the total number of forks of a project. You would also have to add this in that endpoint.

File: apps/api/src/workspace/service/workspace.service.ts [hunk 1]
```diff
@@ -763,7 +763,7 @@ export class WorkspaceService {
     //get all workspaces of user for page with limit
     const items = await this.prisma.workspace.findMany({
       skip: page * limit,
-      take: Math.min(limit, 30),
+      take: Number(limit),
```
Comment (rajdip-b) [inline apps/api/src/workspace/service/workspace.service.ts:766]: This won't be necessary. Just `limit` will do.

Comment (DhairyaMajmudar) [reply apps/api/src/workspace/service/workspace.service.ts]: Done 👍🏻

**Comment by rajdip-b** [comment]
image

You would also need to send in the max limit in here. I would say, create a util function name `limit-max-items-per-page.ts` and call that function across all the `limits`. This way, later on when we need to change the capacity, we only change the function and not all the code.

File: apps/api/src/api-key/service/api-key.service.ts [hunk 2]
```diff
@@ -146,7 +147,7 @@ export class ApiKeyService {
         }
       },
       skip: page * limit,
-      take: limit,
+      take: limitMaxItemsPerPage(limit),
```
Comment (rajdip-b) [inline apps/api/src/api-key/service/api-key.service.ts:150]: Could you please add the pagination response in here? I think the previous contributor missed it.

**Review by rajdip-b — CHANGES_REQUESTED** [review]
For all the files I have the same comments:
- Get rid of `Number()` function
- Replace `limit: Number(limit),` with `limit: limitMaxItemsPerPage(limit),` in every `paginate` function call

**Comment by DhairyaMajmudar** [comment]
> For all the files I have the same comments:
> 
>     * Get rid of `Number()` function
> 
>     * Replace `limit: Number(limit),` with `limit: limitMaxItemsPerPage(limit),` in every `paginate` function call

sure, and I've to do any changes with `page: Number(page),` or that's correct ?

File: apps/api/src/secret/service/secret.service.ts
Comment (rajdip-b) [inline apps/api/src/secret/service/secret.service.ts:1]: There are a few extra newlines in this file.

File: apps/api/src/workspace/service/workspace.service.ts [hunk 3]
```diff
@@ -763,7 +764,7 @@ export class WorkspaceService {
     //get all workspaces of user for page with limit
     const items = await this.prisma.workspace.findMany({
       skip: page * limit,
-      take: Number(limit),
+      take: limit,
```
Comment (rajdip-b) [inline apps/api/src/workspace/service/workspace.service.ts:767]: You missed out on this one

**Review by rajdip-b — CHANGES_REQUESTED** [review]
Replace all occurences of `page: Number(page)` with `page,`

File: apps/api/src/environment/service/environment.service.ts [hunk 4]
```diff
@@ -211,8 +212,8 @@ export class EnvironmentService {
       }
     })
     const metadata = paginate(totalCount, `/environment/all/${projectId}`, {
-      page,
-      limit,
+      page: Number(page),
```
[the remark this answers is not in the document]
Comment (DhairyaMajmudar) [reply apps/api/src/environment/service/environment.service.ts]: Done 👍🏻

File: apps/api/src/secret/service/secret.service.ts
Comment (DhairyaMajmudar) [reply apps/api/src/secret/service/secret.service.ts:1]: Done 👍🏻

File: apps/api/src/workspace/service/workspace.service.ts (hunk 3 above)
Comment (DhairyaMajmudar) [reply apps/api/src/workspace/service/workspace.service.ts]: Done 👍🏻

File: apps/api/src/workspace/service/workspace.service.ts (hunk 1 above)
Comment (rajdip-b) [reply apps/api/src/workspace/service/workspace.service.ts]: You would need to pass ` limitMaxItemsPerPage(limit)` in here :(

**Review by rajdip-b — CHANGES_REQUESTED** [review]
instead of using `page: page`, just use `page,`

**Review by rajdip-b — APPROVED** [review]

**Pull request merged**